src/cgroup.c

```diff
@@ -279,19 +279,29 @@ gboolean check_cgroup2_oom()
 	ssize_t read;
 	while ((read = getline(&line, &len, fp)) != -1) {
 		long int counter;
-
-		if (read < 6 || memcmp(line, "oom ", 4))
+		int memcmp_oom, memcmp_oom_kill, len;
+		const int oom_len = 4, oom_kill_len = 9;
+
+		if (read >= oom_len + 2 && memcmp(line, "oom ", oom_len) == 0)
+			len = oom_len;
+		else if (read >= oom_kill_len + 2 && memcmp(line, "oom_kill ", oom_kill_len) == 0)
```

Comment:
this is a bit pedantic, but why don't we switch these cases? in the oom_kill case, we would always evaluate the first memcmp because `read >= oom_len + 2`, and if we switched them we wouldn't calculate memcmp twice

Reply:
Yes makes sense, switched the if cases.